lib/perl/Genome/Model/Tools/DetectVariants2/Dispatcher.pm

```diff
@@ -1020,38 +1030,69 @@ sub _create_bed_from_vcf {
-                die $self->error_message;
+    
+    for my $variant_type (@{$self->variant_types}) {
+        my $file_accessor = '_'. $variant_type .'_hq_output_file';
+        my $strategy = $variant_type .'_detection_strategy';
+        my $out_dir = $variant_type .'_output_directory';
+        if ( defined($self->$strategy) ) {
+            my $hq_output_dir = $self->output_directory;
+            my $relative_path = $self->get_relative_path_to_result_directory($result->{$out_dir});
+            if ($relative_path) {
+                $hq_output_dir = File::Spec->join($self->output_directory,$relative_path);
+            } else {
+                $self->debug_message('Unable to resolve relative path for '. $variant_type .' output directory. Workflow returned: '. Data::Dumper::Dumper($result));
             }
-            my $hq_output_dir = $self->output_directory."/".$relative_path;
-            my $hq_file;
+
+            my $hq_full_path;
             if ($variant_type eq 'sv' || $variant_type eq 'cnv'){
-                $hq_file = $variant_type."s.hq";
-            }else{
-                $hq_file = $variant_type."s.hq.bed"; # FIXME this will not be true for polymutt or other detectors that output only vcf
-            }
-            my $file;
-            if(-l $hq_output_dir."/".$hq_file){
-                $file = readlink($hq_output_dir."/".$hq_file); # Should look like "dir/snvs_hq.bed"
-                $file = basename($file,['bed']);
-            }
-            else{
-                $file = $hq_file;
+                my $hq_file_name = $variant_type .'s.hq';
+                my $hq_file_path = File::Spec->join($hq_output_dir,$hq_file_name);
+                $hq_full_path = $self->_resolve_variant_file_full_path($hq_output_dir, $hq_file_path);
+                unless ($hq_full_path) {
+                    $hq_full_path = $self->_resolve_variant_file_full_path($hq_output_dir, $hq_file_path .'.vcf.gz');
+                }
+                if ( defined($hq_full_path) ) {
+                    $self->$file_accessor($hq_full_path);
+                } else {
+                    $self->fatal_message('Unable to locate full path to high-quality '. $variant_type .' variant file!');
+                }
+            } else {
+                my $hq_file_name = $variant_type .'s.hq.bed'; # FIXME this will not be true for polymutt or other detectors that output only vcf
+                my $hq_file_path = File::Spec->join($hq_output_dir,$hq_file_name);
+                $hq_full_path = $self->_resolve_variant_file_full_path($hq_output_dir, $hq_file_path);
+                if ( defined($hq_full_path) ) {
+                    $self->$file_accessor($hq_full_path);
+                } else {
+                    $self->debug_message('Unable to locate full path to high-quality '. $variant_type .' variant file. Continuing with \''. $hq_file_path .'\' since DV2 laters resolves the path using the file basename.');
```

Comment:
Just a quick note on this debug message.  I would have thought this should be a fatal message; however, the test 'CopyCatSomaticWithBamWindow.t' will fail if so.  The test is a bit different than most since it has to run the Dispatcher to get the Samtools detector result that is faked in the test.  The /tmp paths are different between the result and the Dispatcher but somehow the test succeeds and the results are generated if the missing path is returned.